src/procstat.c

```diff
@@ -1498,10 +1587,11 @@ void procstat_destroy(struct procstat_context *context)
 		fuse_session_destroy(session);
 	}
 
-	item_put_children_locked(&context->root);
+	item_put_children_locked(&context->root, mutex);
+
 	free(context->mountpoint);
-	pthread_mutex_unlock(&context->global_lock);
-	pthread_mutex_destroy(&context->global_lock);
```

Comment:
So we never destroy our mutexes?

Reply:
It is not possible because we don't know which mutexes are in play at this point. Since the process is going away when this is called, it's not too much of a resource leak.